gtda/homology/point_clouds.py

```diff
@@ -441,19 +429,184 @@ def transform(self, X, y=None):
-            Xt[i], self._homology_dimensions, max_n_points, min_values)
-            for i in range(len(Xt)))
-        Xt = np.stack(Xt)
-        Xt = np.nan_to_num(Xt, posinf=self.infinity_values_)
+        Xt = _postprocess_diagrams(Xt, self._homology_dimensions,
+                                   self.infinity_values_, self.n_jobs)
+        return Xt
+
+
+@adapt_fit_transform_docs
+class EuclideanCechPersistence(BaseEstimator, TransformerMixin):
+    """`Persistence diagrams <https://giotto.ai/theory>`_ resulting from
+    `Cech filtrations <https://giotto.ai/theory>`_.
+    Given a `point cloud <https://giotto.ai/theory>`_ in Euclidean space,
+    information about the appearance and disappearance of topological
+    features (technically, `homology classes <https://giotto.ai/theory>`_) of
+    various dimensions and at different scales is summarised in the
+    corresponding persistence diagram.
+
+xs    Parameters
+    ----------
+    homology_dimensions : iterable, optional, default: ``(0, 1)``
+        Dimensions (non-negative integers) of the topological features to be
+        detected.
+
+    coeff : int prime, optional, default: ``2``
+        Compute homology with coefficients in the prime field
+        :math:`\\mathbb{F}_p = \\{ 0, \\ldots, p - 1 \\}` where
+        :math:`p` equals `coeff`.
+
+    max_edge_length : float, optional, default: ``numpy.inf``
+        Upper bound on the maximum value of the Vietoris-Rips filtration
+        parameter. Points whose distance is greater than this value will
+        never be connected by an edge, and topological features at scales
+        larger than this value will not be detected.
+
+    infinity_values : float or None, default: ``None``
+        Which death value to assign to features which are still alive at
+        filtration value `max_edge_length`. ``None`` means that this death
+        value is declared to be equal to `max_edge_length`.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    infinity_values_ : float
+        Effective death value to assign to features which are still alive at
+        filtration value `max_edge_length`.
+
+    See also
+    --------
+    VietorisRipsPersistence, SparseRipsPersistence, ConsistentRescaling
+
+    Notes
+    -----
+    `GUDHI <https://github.com/GUDHI/gudhi-devel>`_ is used as a C++ backend
+    for computing Cech persistent homology. Python bindings were modified
+    for performance.
+
+    Persistence diagrams produced by this class must be interpreted with
+    care due to the presence of padding triples which carry no information.
+    See :meth:`transform` for additional information.
+
+    References
+    ----------
+    [1] C. Maria, "Persistent Cohomology", 2020; `GUDHI User and Reference \
```

Comment:
I think there need to be indentations, see https://github.com/giotto-ai/giotto-tda/blob/fcffaf8fc9722fc7948d5727fa764ff04258fee2/gtda/homology/point_clouds.py#L95